lib/concurrent/ivar.rb

```diff
@@ -114,14 +121,24 @@ def set(value)
     # @param [Object] reason for the failure
     # @raise [Concurrent::MultipleAssignmentError] if the `IVar` has already
     #   been set or otherwise completed
+    # @return [IVar] self
     def fail(reason = StandardError.new)
-      complete(false, nil, reason)
+      set { raise reason }
```

Comment:
the original `complete(false, nil, reason)` was faster then re-raising in block `set { raise reason }`